url29ff2e53f3_nicolas3470_Settlers_of_Catan_AI_tgz-pJ8-robotGameJ8/src/soc/server/SOCGameTimeoutChecker.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Java Settlers - An online multiplayer version of the game Settlers of Catan
+ * Copyright (C) 2003  Robert S. Thomas
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA  02111-1307, USA.
+ *
+ * The author of this program can be reached at thomas@infolab.northwestern.edu
+ **/
+package soc.server;
+
+
+/**
+ * Wakes up every 5 minutes to check for games that have expired,
+ * by calling {@link SOCServer#checkForExpiredGames()}.
+ *
+ * @author Robert S Thomas
+ */
+public class SOCGameTimeoutChecker extends Thread
+{
+    SOCServer server;
+    boolean alive;
+
+    /**
+     * Create a game timeout checker
+     *
+     * @param srv  the game server
+     */
+    public SOCGameTimeoutChecker(SOCServer srv)
+    {
+        server = srv;
+        alive = true;
+        setName ("timeoutChecker");  // Thread name for debug
+        try { setDaemon(true); } catch (Exception e) {}  // Don't wait on us to exit program
+    }
+
+    /**
+     * Wakes up every 5 minutes to check for games that have expired
+     */
+    public void run()
+    {
+        while (alive)
+        {
+            server.checkForExpiredGames();
+            yield();
+
+            try
+            {
+                // check every 5 minutes; must be at most half of SOCServer.GAME_EXPIRE_WARN_MINUTES
+                sleep(300000);
+            }
+            catch (InterruptedException exc) {}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: InterruptedException is ignored. This can delay thread shutdown and clear the thread’s interrupt status. Only code that implements a thread’s interruption policy can swallow an interruption request.

**Fix**: Rethrow the InterruptedException or reinterrupt the current thread using *Thread.currentThread().interrupt()* so that higher-level interrupt handlers can function correctly.
 If you are wrapping the InterruptedException inside a RuntimeException, call *Thread.currentThread().interrupt()* before throwing the RuntimeException.

Learn more about [interrupts](https://docs.oracle.com/javase/tutorial/essential/concurrency/interrupt.html) and [dealing with InterruptedException](https://www.ibm.com/developerworks/java/library/j-jtp05236/index.html?ca=drs-#N10187)